service/worker/archiver/activities.go

```diff
@@ -244,11 +258,27 @@ func uploadBlob(ctx context.Context, blobstoreClient blobstore.Client, bucket st
 	return nil
 }
 
+func downloadBlob(ctx context.Context, blobstoreClient blobstore.Client, bucket string, key blob.Key) (*blob.Blob, error) {
+	bCtx, cancel := context.WithTimeout(ctx, blobstoreTimeout)
```

Comment:
I'm a little worried that we do retries in a tight loop without backoff.  Is there a reason you are not using our regular retry abstraction for retries?

Reply:
The blobstore client itself is a retryable blobstore client. So each call to Download will kick off a series of retries which have backoff between them. Is this acceptable or do you want me to use the retryable client inside of an operation passed to Retry function?